File: leetcode/string/2011.py
from typing import List
from queue import Queue
class Solution:
    def finalValueAfterOperations(self, operations: List[str]) -> int:
        res:int = 0
        for operation in operations:
            if operation[0] == '-' or operation[2] == '-':
                res -= 1
            else:
                res += 1
        return res
if __name__ == "__main__":
    solution = Solution()
    operations =["--X","X++","X++"]
    result = solution.finalValueAfterOperations(operations)
    print(f"result = {result}")
 

File: leetcode/string/2011.cpp
#include <algorithm>
#include <iostream>
#include <unordered_set>
#include <stack>
#include <queue>
#include <vector>
#include <functional> //包含std::greater 和 std::less
#include <math.h>
#include <limits.h>
using namespace std;
struct TreeNode {
   int val;
   TreeNode *left;
   TreeNode *right;
   TreeNode() : val(0), left(nullptr), right(nullptr) {}
   TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
   TreeNode(int x, TreeNode *left, TreeNode *right)
       : val(x), left(left), right(right) {}
};
struct ListNode {
   int val;
   ListNode *next;
   ListNode() : val(0), next(nullptr) {}
   ListNode(int x) : val(x), next(nullptr) {}
   ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
  int finalValueAfterOperations(vector<string> &operations) {
   int res = 0;
   for(string operation : operations){
     if (operation[0] == '-' || operation[2] == '-'){
      res --;
     }else{
      res ++;
     }
   }
   return res;
  }
};
int main() {
   // 示例二叉树
   Solution solution;
   vector<string> operations = {"--X", "X++", "X++"};
   int result = solution.finalValueAfterOperations(operations);
   std::cout << "result: " << result << std::endl;
}